V6.1.0 rc (#3744)
kframework/k: CHANGELOG.md

copyright: Copyright (c) K Team. All Rights Reserved.
---

K Framework 6.1.0
=================

Features
--------

> - Added support for MacOS 13 Ventura and Debian 12 Bookworm. We dropped support for

Reviewer: ```suggestion
- Added support for MacOS 13 Ventura. We dropped support for
```

>   Ubuntu 20 Focal. You can now build K from sources on Apple Silicon. See README

Reviewer: ```suggestion
  Ubuntu 20.04 Focal. K can now be built from source on Apple Silicon. See README
```

  for more details.

- Updated dependency to Java version 17 or higher.

- Added the Haskell Backend Booster as a dependency to K. This can improve performance

>   when running large proofs. It combines the llvm-backend for concrete execution and

Reviewer: ```suggestion
  when running large proofs. It uses the LLVM Backend for concrete execution and
```

>   relies on the haskell-backend to simplify terms when there is a split in the proof.

Reviewer: ```suggestion
  relies on the Haskell Backend to simplify terms when there is a split in the proof.
```

- Optimized the kompiler by removing unit applications for collections.

- Minimize JSON output by dropping unused attributes

- Added `--smt-timeout` flag to `krun` and `kprove`.

- Changed the Maven repository to Cloudrepo for more stability and flexibility when
  building K from sources.

- Improved attribute error messages by creating a whitelist dependent on the context.
  This check is now mandatory.

> - Rule label cannot contain ` or whitespace

Reviewer: ```suggestion
- Rule label can no longer contain backticks (`) or whitespace
```

- Improved the help messages by adding a description of the expected parameter.

- Documentation: Started work on Section 2 of the tutorial. Added a description for
  `kserver`.

- Added `--debugger-command` flag to krun.

- Added two options `--debug-tokens` and `--debug-parse` to help with debugging
  parsing errors. The first option will print a Markdown table with all the matched
  tokens by the scanner. The second one will give more details about the partial parse
  tree constructed before an error was encountered.

Misc/Bug Fixes
--------------

> - Fixed a bug where the llvm backend would segfault because of badly initialized

Reviewer: ```suggestion
- Fixed a bug where the LLVM backend would segfault because of badly initialized
```
